### DISCUSSION TITLE
Generating Animation Keypaths when layers have "." in their name

### DISCUSSION THREAD

**Discussion by daniel-e-gonzalez** [airbnb/lottie-ios#2457] [General]
Hi, I have a lottie animation that has layers that have `.` in their name. This is causing issues with being able to generate `AnimationKeypaths` from an `allHierarchyKeypaths` call on my `LottieAnimationView` since the default `AnimationKeypath` constructor delimits by `.`. The generated `AnimationKeypath` has an entry of "" and then another entry for "myLayerName". It really should have a single entry for ".myLayerName".

I am able to work around this by using `AnimationKeypath(keys: myKeys)` (where `myKeys` is generated using some workaround code), but I would like to instead be able to just use `AnimationKeypath(keypath: keypath)`. Can this be fixed?

## Which Version of Lottie are you using?

Lottie 4.5.0

## Expected Behavior 
`AnimationKeypath(keypath: keypath)` returns the correct keypaths when layers have `.` in the name.

## Actual Behavior
There is a blank entry and an incorrectly named entry in the keypath.

**Comment by calda** [comment]
We could make it so `\.` escapes the `.` to be a part of the layer name rather than a separator, but this risks breaking existing use cases of `\` in key paths. I think using `AnimationKeypath(keys:)` is a pretty reasonable workaround.

**Reply by daniel-e-gonzalez** [reply]
Got it. I'm ok with using the workaround, but just out of curiosity, is it ever valid for there to be an entry in the `AnimationKeypath` that is just `""`? If it isn't, it could be possible for the Lottie library to detect if this edge case has occurred and adjust it accordingly.  

Thanks for the quick responses on all of this!

**Reply by daniel-e-gonzalez** [reply]
ah, sorry i just realized that only happens in my specific case where my layer name _begins_ with "." so not really a great rule it can use i suppose :/
